codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ServerUpgradeCodec.java

```diff
@@ -94,7 +95,11 @@ public void prepareUpgradeResponse(ChannelHandlerContext ctx, FullHttpRequest up
                         + HTTP_UPGRADE_SETTINGS_HEADER + " header.");
             }
             Http2Settings settings = decodeSettingsHeader(ctx, upgradeHeaders.get(0));
-            connectionHandler.onHttpServerUpgrade(settings);
+            if (connectionHandler instanceof Http2ConnectionHandler) {
+                ((Http2ConnectionHandler) connectionHandler).onHttpServerUpgrade(settings);
+            } else {
```

Comment:
should we have an explicit `else if ( instanceof Http2MultiplexCodec) {..} else { /*throw descriptive exception*/ }`?